simplemma/language_detector.py

```diff
@@ -91,10 +111,13 @@ def lang_detector(
     results = sorted(myresults.items(), key=itemgetter(1), reverse=True)
     # post-processing
     if len(results) > 1:
-        # in case of ex-aequo
-        if greedy is False and results[0][1] == results[1][1]:
-            results = lang_detector(text, lang=lang, greedy=True)
-        # fallback
-        if len(results) > 1 and results[0][1] == results[1][1]:
-            return _return_default()
+        # switch unknown first item to the end
+        if results[0][0] == "unk":
+            pair = results.pop(0)
+            results.append(pair)
+        # in case of ex-aequo use other token sampling to discriminate
+        if not greedy and results[0][1] == results[1][1]:
+            results = lang_detector(
+                text, lang=lang, greedy=True, token_sampler=RelaxedTokenSampler()
```

Comment:
I'm a bit conflicted here.
We let the user provide a `TokenSampler`. But then there is a case in which we hardcode a separate TokenSampler.

Hardcoded things remove flexibility in general.

I would say that we should return whatever the TokenSampler produces even if there is a tie.
We could document, that in case of a tie, you can try again a different TokenSampler (and document how to configure the RelaxedTokenSampler as I suggested above)